pkg/controller/applicationmonitoring/applicationmonitoring_controller.go

```diff
@@ -362,6 +363,36 @@ func (r *ReconcileApplicationMonitoring) reconcileBlackboxExporterConfig(cr *app
 	if r.extraParams == nil {
 		r.extraParams = map[string]string{}
 	}
+
+	// Add bearer_token to ConfigMap to allow Blackbox exporter requests to get through proxy in front of Grafana service
+	blackboxServiceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "blackbox-exporter-service-account",
+			Namespace: "redhat-rhoam-middleware-monitoring-operator",
+		},
+	}
+	if err := r.client.Get(ctx, client.ObjectKey{Name: blackboxServiceAccount.Name, Namespace: blackboxServiceAccount.Namespace}, blackboxServiceAccount); err != nil {
+		log.Error(err, "client.Get")
+		return fmt.Errorf("error getting blackbox exporter service account: %s", err.Error())
+	}
+	var secretName string
+	for _, secret := range blackboxServiceAccount.Secrets {
+		if res, _ := regexp.MatchString("blackbox-exporter-service-account-token", secret.Name); res {
+			secretName = secret.Name
+		}
+	}
+	blackboxServiceAccountSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretName,
+			Namespace: "redhat-rhoam-middleware-monitoring-operator",
```

Comment:
Same as above, namespace shouldn't be hard-coded.
Here I would reference `blackboxServiceAccount.Namespace`

Reply:
Fixed